api/v1alpha1/clusterextension_types.go

```diff
@@ -73,8 +73,41 @@ type ClusterExtensionSpec struct {
 	// the bundle may contain resources that are cluster-scoped or that are
 	// installed in a different namespace. This namespace is expected to exist.
 	InstallNamespace string `json:"installNamespace"`
+
+	//+kubebuilder:Optional
+	// Preflight defines the strategy for the preflight checks (i.e. as of now, the CRD upgrade safety checks) to be applied or skipped when attempting to install the cluster extension.
+	Preflight PreflightConfig `json:"preflight,omitempty"`
+}
+
+// PreflightConfig holds the config for the preflight checks. Currently, this is implemented for the
+// CRDUpgradeSafety preflight check and can be extended in the future to add other preflight checks.
+type PreflightConfig struct {
+	CRDUpgradeSafety CRDUpgradeSafetyPreflightConfig `json:"crdUpgradeSafety,omitempty"`
+}
+
+// CRDUpgradeSafetyPreflightConfig is a custom struct that holds the necessary configuration values to
+// enable or disable the CRD Upgrade Safety validations. Currently, this holds Mode
+// that sets the CRD Upgrade Safety validations to either Enabled/Disabled.
+// By default, this is set to "Enabled".
+type CRDUpgradeSafetyPreflightConfig struct {
+	//+kubebuilder:validation:Enum:=Enabled;Disabled
+	//+kubebuilder:default:=Enabled
+	//+kubebuilder:Optional
+	Mode Mode `json:"mode,omitempty"`
 }
 
+type Mode string
+
+const (
+	// CRDUpgradeSafetyPreflightConfigEnabled represents the default state for the
+	// CRD Upgrade Safety validations by setting it to "Enabled".
+	CRDUpgradeSafetyPreflightConfigEnabled Mode = "Enabled"
+
+	// CRDUpgradeSafetyPreflightConfigDisabled represents the option for the
+	// CRD Upgrade Safety validations to be disabled by setting it to "Disabled".
+	CRDUpgradeSafetyPreflightConfigDisabled Mode = "Disabled"
+)
```

Comment:
Instead of this being a more generic type called `Mode`, what do you think about making this type more unique to each preflight check we add? For this case maybe something like:
```suggestion
type CRDUpgradeSafetyMode string

const (
	// CRDUpgradeSafetyModeEnabled represents the default state for the
	// CRD Upgrade Safety validations by setting it to "Enabled".
	CRDUpgradeSafetyModeEnabled Mode = "Enabled"

	// CRDUpgradeSafetyModeDisabled represents the option for the
	// CRD Upgrade Safety validations to be disabled by setting it to "Disabled".
	CRDUpgradeSafetyModeDisabled Mode = "Disabled"
)
```

My thinking here is that this more easily allows us to manage/maintain a unique set of modes for each preflight check. I could totally be wrong here though and this is just a suggestion - let me know what you think!

Maybe @joelanford also has some opinions here?

Reply:
I intentionally had it as `Mode` as it's going to be a property within the CRD Upgrade Safety field so it felt like it may be redundant to have it as something like `CRDUpgradeSafetyMode`.
So for example, let's say we add another preflight check called `ClusterHealthValidation`, below is how the `ClusterExtension` yaml would look like:

```              
              preflight:
                properties:
                  clusterHealthValidation:
                    properties:
                      mode:
                        default: Enabled
                        enum:
                        - Enabled
                        - Disabled
                        type: string
                    type: object
                  crdUpgradeSafety:
                    properties:
                      mode:
                        default: Enabled
                        enum:
                        - Enabled
                        - Disabled
                        type: string
                    type: object
                type: object
```

But if it helps to differentiate easily and visually, then I think changing it to `CRDUpgradeSafetyMode` to be more specific would be better. My opinion is that it may be a bit redundant, but I am open to changing it if you feel it's better that way.

Also, I like the shortened names: `CRDUpgradeSafetyModeEnabled` and `CRDUpgradeSafetyModeDisabled`. :+1:

Reply:
I'm fine with a single `Mode` type as long as there is an easy way to differentiate which preflight checks can use which modes. If we are going with a single type, maybe something like `PreflightMode` is a bit more descriptive?

Reply:
Addressed in [latest commit](https://github.com/operator-framework/operator-controller/pull/845/commits/224d1393d317567db4c1f17208d31dd444772080) based on your original suggestion as I like it better and it wouldn't hurt to be specific :)